yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -322,7 +322,7 @@ private[spark] class Client(
             destName.orElse(Option(localURI.getFragment())).getOrElse(localPath.getName())
           val destPath = copyFileToRemote(dst, localPath, replication)
           distCacheMgr.addResource(
-            fs, hadoopConf, destPath, localResources, resType, linkname, statCache,
```

Comment:
i think we can replace fs with destPath's FileSystem. like: 
val fs = FileSystem.get(destPath.toUri(), hadoopConf)